flinkprocessor/src/main/java/io/pravega/example/videoprocessor/FlinkFaceRecognizerJob.java

```diff
@@ -0,0 +1,293 @@
+            final DataStream<VideoFrame> videoFrames = inChunkedVideoFramesWithTimestamps
+                    .map(VideoFrame::new)
+                    .uid("ChunkedVideoFrameReassembler")
+                    .name("ChunkedVideoFrameReassembler");
+
+            //
+            // Create non-video person database transaction datastream.
+            //
+
+            final FlinkPravegaReader<Transaction> flinkPravegaReaderTransactions = FlinkPravegaReader.<Transaction>builder()
+                    .withPravegaConfig(getConfig().getPravegaConfig())
+                    .forStream(getConfig().getPersonDatabaseStreamConfig().getStream())
+                    .withDeserializationSchema(new JsonDeserializationSchema<>(Transaction.class))
+                    .build();
+            final DataStream<Transaction> personDatabaseTransactions = env
+                    .addSource(flinkPravegaReaderTransactions)
+                    .uid("transaction")
+                    .name("transaction");
+            personDatabaseTransactions.printToErr().uid("personDatabaseTransactions-print").name("personDatabaseTransactions-print");
+
+            // Assign timestamps and watermarks based on timestamp in each chunk.
+            final DataStream<Transaction> transactionsWithTimestamps = personDatabaseTransactions
+                    .assignTimestampsAndWatermarks(
+                            new BoundedOutOfOrdernessTimestampExtractor<Transaction>(
+                                    Time.milliseconds(getConfig().getMaxOutOfOrdernessMs())) {
+                                @Override
+                                public long extractTimestamp(Transaction element) {
+                                    return element.timestamp.getTime();
+                                }
+                            })
+                    .uid("assignTimestampsAndWatermarksSensor")
+                    .name("assignTimestampsAndWatermarksSensor");
+            transactionsWithTimestamps.printToErr().uid("transactionsWithTimestamps-print").name("transactionsWithTimestamps-print");
+
+            // Calculate embeddings for image data in transactions
+            final DataStream<Transaction> transactionEmbeddings = transactionsWithTimestamps
+                    .map(new FaceRecognizerMapFunctionTransaction())
+                    .uid("TransactionEmbeddings")
+                    .name("transactionEmbeddings");
+
+            // For each camera and window, get the most recent frame.
+            final DataStream<VideoFrame> lastVideoFramePerCamera = videoFrames
+                    .keyBy((KeySelector<VideoFrame, Integer>) value -> value.camera)
+                    .window(TumblingEventTimeWindows.of(Time.milliseconds(periodMs)))
+                    .maxBy("timestamp")
+                    .uid("lastVideoFramePerCamera")
+                    .name("lastVideoFramePerCamera");
+
+            // This is calculating the embeddings for the located face in the videoframe.
```

Comment:
Change "face" to "faces" to indicate that this will handle multiple faces in a video frame.